### PR TITLE
feat(input): Build Text Input Styles

### PR DESCRIPTION
**Pull request by ssalinas24**
JIRA: https://jira.rax.io/browse/SURF-512 

### LGTM's
- [x] Dev LGTM
- [x] Design LGTM
- [x] Zoom LGTM

## Text Input
01_text-input

## Email
02_email

## Password
03_password

## Text Area
04_text-area

## CSS Classes
05_classes

### CODE REVIEW

File: source/components/text-input/index.html [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+---
+{% extends 'component.njk' %}
+{% block content %}
+
+<section>
+  <h2 class="hxSectionTitle" id="text">Text</h2>
+  <div class="demo">
+    <form>
+      <div class="hxRow">
+        <div class="hxCol">
```
Comment (CITguy) [inline source/components/text-input/index.html]: Are the `.hxRow` and `.hxCol` needed?

File: source/components/text-input/index.html [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+          <div class="hxFormGroup">
+            <label for="txtDisabled">Disabled</label>
+            <input type="text" id="txtDisabled" placeholder="Text" disabled>
+          </div>
+        </div>
+      </div>
+    </form>
+  </div>
+  {% code 'html' %}
+  <div class="hxFormGroup">
```
Comment (CITguy) [inline source/components/text-input/index.html]: I'd wrap these in `<form>`. Also, please indent.

File: source/components/text-input/index.html [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+          <div class="hxFormGroup">
+            <label for="txtDisabledEmail">Disabled</label>
+            <input type="email" id="txtDisabledEmail" placeholder="Email" disabled>
+          </div>
+        </div>
+      </div>
+    </form>
+  </div>
+  {% code 'html' %}
+  <div class="hxFormGroup">
```
Comment (CITguy) [inline source/components/text-input/index.html]: Please wrap in a `<form>` and indent.

File: source/components/text-input/index.html [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  </div>
+  {% endcode %}
+</section>
+
+<section>
+  <h2 class="hxSectionTitle" id="email">Email</h2>
+  <div class="demo">
+    <form>
+      <div class="hxRow">
+        <div class="hxCol">
```
Comment (CITguy) [inline source/components/text-input/index.html]: I don't think `.hxRow` and `.hxCol` are needed.

File: source/components/text-input/index.html [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+  </div>
+  {% endcode %}
+</section>
+
+<section>
+  <h2 class="hxSectionTitle" id="password">Password</h2>
+  <div class="demo">
+    <form>
+      <div class="hxRow">
+        <div class="hxCol">
```
Comment (CITguy) [inline source/components/text-input/index.html]: I don't think `.hxRow` and `.hxCol` are needed.

File: source/components/text-input/index.html [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+          <div class="hxFormGroup">
+            <label for="txtDisabledPassword">Disabled</label>
+            <input type="password" id="txtDisabledPassword" placeholder="Password" disabled>
+          </div>
+        </div>
+      </div>
+    </form>
+  </div>
+  {% code 'html' %}
+  <div class="hxFormGroup">
```
Comment (CITguy) [inline source/components/text-input/index.html]: Please wrap in `<form>` and indent.

File: source/components/text-input/index.html [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+  </div>
+  {% endcode %}
+</section>
+
+<section>
+  <h2 class="hxSectionTitle" id="text-area">Text Area</h2>
+  <div class="demo">
+    <form>
+      <div class="hxRow">
+        <div class="hxCol">
```
Comment (CITguy) [inline source/components/text-input/index.html]: I don't think `.hxRow` and `.hxCol` are needed.

**Comment by ssalinas24** [comment]
PR Feedback has been addressed.

File: source/components/text-input/index.html [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+  {% endcode %}
+</section>
+
+<section>
+  <h2 class="hxSectionTitle" id="css-classes">CSS Classes</h2>
+
+  <dl>
+    <dt><code>.-hxInvalid</code></dt>
+    <dd>
+      Use on a form input element to apply Helix styles.
```
Comment (CITguy) [inline source/components/text-input/index.html]: Description can be shortened to 

```html
<dd>Add to form input to apply invalid styling.</dd>
```

File: source/components/text-input/text-input.less [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  &::-webkit-input-placeholder { // Chrome
+    color: @gray-600;
+  }
+
+  &::-moz-placeholder { // Firefox 19+
+    color: @gray-600;
+    opacity: 1;
+  }
+
+  &:-moz-placeholder { // Firefox 18-
```
Comment (CITguy) [inline source/components/text-input/text-input.less]: We can probably remove this block given that we don't support versions of Firefox that old.

File: source/components/text-input/text-input.less [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  &.-hxInvalid {
+    border: 2px solid @red-900;
+  }
+
+  &[disabled] {
+    background-color: @gray-50;
+    border-color: @gray-300;
+    color: @gray-400;
```
[the remark this answers is not in the document]
Comment (CITguy) [reply source/components/text-input/text-input.less]: updated below

**Comment by technabors** [comment]
Giving the Design LGTM

**Comment by ssalinas24** [comment]
Feedback Addressed again

**Pull request merged**